data_model/src/block_value.rs

```diff
@@ -0,0 +1,82 @@
+//! This module contains [`BlockValue`] and [`BlockHeaderValue`] structures, their implementation and related traits and
+//! instructions implementations.
+#[cfg(not(feature = "std"))]
+use alloc::Vec;
+use core::cmp::Ordering;
+
+use iroha_crypto::{Hash, HashOf, MerkleTree};
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    events::Event,
+    transaction::{TransactionValue, VersionedTransaction},
+};
+
+/// Block header
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+pub struct BlockHeaderValue {
+    /// Unix time (in milliseconds) of block forming by a peer.
+    pub timestamp: u128,
+    /// a number of blocks in the chain up to the block.
+    pub height: u64,
+    /// Hash of a previous block in the chain.
+    /// Is an array of zeros for the first block.
+    pub previous_block_hash: Hash,
+    /// Hash of merkle tree root of the tree of valid transactions' hashes.
+    pub transactions_hash: HashOf<MerkleTree<VersionedTransaction>>,
+    /// Hash of merkle tree root of the tree of rejected transactions' hashes.
+    pub rejected_transactions_hash: HashOf<MerkleTree<VersionedTransaction>>,
+    /// Hashes of the blocks that were rejected by consensus.
+    pub invalidated_blocks_hashes: Vec<Hash>,
```

Comment:
Since I can't use `VersionedCommitedBlock` here and `HashOf<BlockValue>` would be different to `HashOf<VersionedCommitedBlock>` it's just naked `Hash`. I don't think there's any good solution to be found without moving definition of blocks to separate crate entirely.

Reply:
yes, actually blocks should be moved to the `data_model` crate from the `core` crate. Because as far as I can see you are duplicating the block definition

Reply:
while this is what I would like to see the most I'm not sure everyone would agree

Reply:
I think that Block -> data_model is beyond the scope of a last minute change.